packages/core/src/api/Editor.ts

```diff
@@ -0,0 +1,433 @@
+
+    this.tiptapEditor.commands.BNUpdateBlock(posBeforeNode + 1, updatedBlock);
+  }
+
+  /**
+   * Removes multiple blocks from the editor. Throws an error if any of the blocks could not be found.
+   * @param blocksToRemove An array of blocks that should be removed.
+   */
+  public removeBlocks(blocksToRemove: Block[]) {
+    const idsOfBlocksToRemove = new Set<string>(
+      blocksToRemove.map((block) => block.id)
+    );
+
+    let removedSize = 0;
+
+    this.tiptapEditor.state.doc.descendants((node, pos) => {
+      // Skips traversing nodes after all target blocks have been removed.
+      if (idsOfBlocksToRemove.size === 0) {
+        return false;
+      }
+
+      // Keeps traversing nodes if block with target ID has not been found.
+      if (
+        node.type.name !== "blockContainer" ||
+        !idsOfBlocksToRemove.has(node.attrs.id)
+      ) {
+        return true;
+      }
+
+      idsOfBlocksToRemove.delete(node.attrs.id);
+      const oldDocSize = this.tiptapEditor.state.doc.nodeSize;
+
+      this.tiptapEditor.commands.BNDeleteBlock(pos - removedSize + 1);
+
+      const newDocSize = this.tiptapEditor.state.doc.nodeSize;
+      removedSize += oldDocSize - newDocSize;
+
+      return false;
+    });
+
+    if (idsOfBlocksToRemove.size > 0) {
+      let notFoundIds = "";
+
+      for (const id of Array.from(idsOfBlocksToRemove)) {
+        notFoundIds += "\n" + id;
+      }
+
+      throw Error(
+        "Blocks with the following IDs could not be found in the editor: " +
+          notFoundIds
+      );
+    }
+  }
+
+  /**
+   * Replaces multiple blocks in the editor with several other blocks. If the provided blocks to remove are not adjacent
+   * to each other, the new blocks are inserted at the position of the first block in the array. Throws an error if any
+   * of the blocks could not be found.
+   * @param blocksToRemove An array of blocks that should be replaced.
+   * @param blocksToInsert An array of blocks to replace the old ones with.
+   */
+  public replaceBlocks(
+    blocksToRemove: Block[],
+    blocksToInsert: PartialBlock[]
+  ) {
+    this.insertBlocks(blocksToInsert, blocksToRemove[0]);
+    this.removeBlocks(blocksToRemove);
+  }
+
+  /**
+   * Executes a callback function whenever the editor's content changes.
+   * @param callback The callback function to execute.
+   */
+  public onContentChange(callback: () => void) {
+    this.tiptapEditor.on("update", callback);
+  }
+
+  /**
+   * Serializes a list of blocks into an HTML string. The output is not the same as what's rendered by the editor, and
+   * is simplified in order to better conform to HTML standards. Block structuring elements are removed, children of
+   * blocks which aren't list items are lifted out of them, and list items blocks are wrapped in `ul`/`ol` tags.
+   * @param blocks The list of blocks to serialize into HTML.
+   */
+  public async blocksToHTML(blocks: Block[]): Promise<string> {
```

Comment:
I think it's worth considering splitting this file further apart, and for example extracting these functions to separate files (one for html conversions, one for markdown conversions, etc)